seerep_srv/seerep_server/src/fb_cameraintrinsics_service.cpp

```diff
@@ -68,6 +68,48 @@ grpc::Status FbCameraIntrinsicsService::TransferCameraIntrinsics(
 
   BOOST_LOG_SEV(m_logger, boost::log::trivial::severity_level::info) << "received camera instrinsics for storage ";
 
+  // check if the distortion model is found in kCameraDistortionModels
+  bool distortion_model_err =
+      requestRoot->distortion_model()->size() > 0 &&
+      std::find(std::begin(seerep_server_constants::kCameraDistortionModels),
+                std::end(seerep_server_constants::kCameraDistortionModels),
+                requestRoot->distortion_model()->c_str()) == std::end(seerep_server_constants::kCameraDistortionModels);
+
+  bool intrinsics_mat_err = !requestRoot->intrinsic_matrix() || requestRoot->intrinsic_matrix()->size() != 9 ||
+                            (*requestRoot->intrinsic_matrix())[0] == 0 || (*requestRoot->intrinsic_matrix())[4] == 0;
+
+  bool rectification_mat_err =
+      requestRoot->rectification_matrix() &&
+      (requestRoot->rectification_matrix()->size() > 0 && requestRoot->rectification_matrix()->size() != 9);
+
+  bool projection_mat_err = requestRoot->projection_matrix() && (requestRoot->projection_matrix()->size() > 0 &&
+                                                                 requestRoot->projection_matrix()->size() != 12);
+
+  if (distortion_model_err || intrinsics_mat_err || rectification_mat_err || projection_mat_err)
+  {
+    response_message = "";
+    if (distortion_model_err)
+    {
+      response_message += "The distortion model is set but invalid. ";
+    }
+    if (intrinsics_mat_err)
+    {
+      response_message += "Instrinsics matrix is not set, dimensions are incorrect or the focal length values are "
+                          "invalid. ";
+    }
+    if (rectification_mat_err)
+    {
+      response_message += "Rectification matrix dimensions are incorrect. ";
+    }
+    if (projection_mat_err)
+    {
+      response_message += "Projection matrix dimensions are incorrect.";
+    }
+    BOOST_LOG_SEV(m_logger, boost::log::trivial::severity_level::error) << response_message;
+    seerep_server_util::createResponseFb(response_message, seerep::fb::TRANSMISSION_STATE_FAILURE, response);
+    return grpc::Status(grpc::StatusCode::INVALID_ARGUMENT, response_message);
+  }
```

Comment:
The checks for Protocol Buffers and Flatbuffers look very similar and could be combined. However, we currently don't have a common place to share this implementation. I think it's fine for now, but we should keep this in mind.